src/request_queue.js

```diff
@@ -197,26 +206,45 @@ export class RequestQueue {
         checkParamOrThrow(queueName, 'queueName', 'Maybe String');
         checkParamOrThrow(clientKey, 'clientKey', 'String');
 
-        if (!clientKey) throw new Error('Parameter "clientKey" must be nonempty string!');
+        if (!clientKey) throw new Error('Parameter "clientKey" must be a non-empty string!');
 
         this.clientKey = clientKey;
         this.queueId = queueId;
         this.queueName = queueName;
+
+        // Contains a cached list of request IDs from the head of the queue,
+        // as obtained in the last query. Both key and value is the request ID.
         this.queueHeadDict = new ListDictionary();
```

Comment:
If that's what it does, maybe we should use a `Set` then? Natives are generally well performance optimized.

Reply:
good point, will change it to Set

Reply:
Actually, we need ListDictionary here because we need to manage the order of items, which in supported by Set. But I will use it for `inProgress`

Reply:
You can iterate a `Set` in insertion order.